src/Microsoft.AspNetCore.Mvc.TagHelpers/TagHelperOutputExtensions.cs

```diff
@@ -154,6 +154,93 @@ public static void MergeAttributes(this TagHelperOutput tagHelperOutput, TagBuil
             }
         }
 
+        /// <summary>
+        /// Adds the given <paramref name="classValue"/> to the <paramref name="tagHelperOutput"/>'s
+        /// <see cref="TagHelperOutput.Attributes"/>.
+        /// </summary>
+        /// <param name="tagHelperOutput">The <see cref="TagHelperOutput"/> this method extends.</param>
+        /// <param name="classValue">The class value to add.</param>
+        public static void AddClass(
+            this TagHelperOutput tagHelperOutput,
+            string classValue)
+        {
+            if (tagHelperOutput == null)
+            {
+                throw new ArgumentNullException(nameof(tagHelperOutput));
+            }
+
+            if (!tagHelperOutput.Attributes.ContainsName("class"))
+            {
+                tagHelperOutput.Attributes.Add("class", classValue);
+            }
+            else
+            {
+                var classAttributeValue = tagHelperOutput.Attributes["class"].Value.ToString();
+                if (classAttributeValue.Contains(" "))
+                {
+                    var arrayOfClasses = classAttributeValue.Split(' ');
+                    if (arrayOfClasses.Contains(classValue))
+                    {
+                        return;
+                    }
+                }
+
+                if (!classAttributeValue.Equals(classValue))
```

Comment:
I'm just guessing, but if the string is exactly "foo" and you're splitting on spaces, you'll get an array of 1 element with the string "foo", so it would work, no? I think adding a unit test for this case would prove it one way or the other 😄